plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java

```diff
@@ -149,7 +158,24 @@ public void execute(DockerClient dockerClient)
     }
 
     writeMetadata(log);
-
+    
+    if (saveImageToTarArchive != null && imageId != null) {
+      try {
+        log.info(MessageFormat.format("Image will be saved as a file {0}", 
+            Paths.get(saveImageToTarArchive).toString()));
+        final InputStream is = dockerClient.save(imageId);
+        java.nio.file.Files.copy(is, Paths.get(saveImageToTarArchive),
+            StandardCopyOption.REPLACE_EXISTING);
+      } catch (IOException e) {
+        log.warn("IOException thrown while trying to write the image archive.");
+      } catch (DockerException e) {
+        log.warn("Docker threw an exception while trying to write the image archive: "
+            + e.getMessage());
+      } catch (InterruptedException e) {
+        log.warn("InterruptedException thrown while trying to write the image archive.");
+      }
+    }
```

Comment:
I'm curious about the error handling here. It seems like if an exception was thrown by `dockerClient.save()` (or the file copy) then the plugin execution will continue, and probably print a `BUILD SUCCESS` message from Maven, but with a warning message that saving the image archive to a tarball failed.

How about failing the build in those cases? I'd assume that if someone configured the build to save the image archive to a file, they'd want the build to fail if that action failed.

Reply:
Makes perfect sense to me. I already found one case where an exception occurs: running the `basic` integration test, The image has `scratch` as a base so the tar archive is 0 bytes and an exception is thrown.

Do you think an extra integration test is needed to test the archive export 
as well?

Reply:
> Do you think an extra integration test is needed to test the archive export
> as well?

I think that would be nice to have but not necessary